HR_allfuncs.py

```diff
@@ -118,149 +122,155 @@ def HR_peakdetect(data_array):
+        averaging_time_sec = averaging_time * min_to_sec
 
-    # extract hr data from .csv file
-    max_acq_time = max(time_array)
+        # extract hr data from .csv file
+        max_acq_time = max(time_array)
 
-    # check if averaging time is longer than ECG acq time
-    if averaging_time_sec > max_acq_time:
-        print("Your averaging time is longer than the ECG acquisition time, try a new value")
-        raise ValueError
-    else:
-        pass
+        # check if averaging time is longer than ECG acq time
+        if averaging_time_sec > max_acq_time:
+            print("Your averaging time is longer than the ECG acquisition time, try a new value")
+            raise ValueError
+        else:
+            pass
 
 
-    final_ind = 0
-    final_min = abs(time_array[0] - averaging_time_sec)
-    # XXX = []
-    # test_pos = np.argwhere(min(abs(time_array - averaging_time_sec)))
-    # print(test_pos)
+        final_ind = 0
+        final_min = abs(time_array[0] - averaging_time_sec)
+        # XXX = []
+        # test_pos = np.argwhere(min(abs(time_array - averaging_time_sec)))
+        # print(test_pos)
 
-    for i in range(0, len(time_array)):
+        for i in range(0, len(time_array)):
 
-        min_val = abs(time_array[i] - averaging_time_sec)
-        if min_val < final_min:
-            final_min = min_val
-            final_ind = i
+            min_val = abs(time_array[i] - averaging_time_sec)
+            if min_val < final_min:
+                final_min = min_val
+                final_ind = i
 
-    # final_ind = np.argwhere(min(abs(time_array - averaging_time_sec)))
-    # avg_index = (np.abs(time_array - averaging_time_sec)).argmin()
+        # final_ind = np.argwhere(min(abs(time_array - averaging_time_sec)))
+        # avg_index = (np.abs(time_array - averaging_time_sec)).argmin()
 
-    time_array_sliced = time_array[:final_ind+1]
-    average_hr_val = int(round((len(time_array_sliced))/averaging_time))
-    return average_hr_val
+        time_array_sliced = time_array[:final_ind+1]
+        self.avg_hr_val = int(round((len(time_array_sliced))/averaging_time))
+        dt_first_beat = time_array[2] - time_array[1]
+        self.inst_hr_val = min_to_sec  * 1 / dt_first_beat
 
 
-def tachy(average_hr_val, tachy_limit):
 
-    """
-    .. function:: tachy(average_hr_val, tachy_limit)
+class Diagnosis:
 
-    Determine if tachycardia occured during ECG acquisition.
 
-    :param average_hr_val: average HR value calculated from hr_averaging()
-    :param tachy_limit: tachycardia limit to be specified
-    :rtype: bool (True if tachycardia is not present, False if tachycardia is present)
-    """
+    def __init__(self):
+        self.tachy_result = bool
+        self.brachy_result = bool
 
-    # check for non-zero averaging time
-    if tachy_limit <= 0:
-        print("Your tachycardia limit must be greater than zero.")
-        raise ValueError
+    def tachy(self, average_hr_val, tachy_limit=100):
```

Comment:
nice! This function could probably check a flag to see if the tachy_result was already calculated on non-stale data and could then immediately return that without redoing the calc.

The necessity to pass in average_hr_val may complicate that, but if you're up for it you may be able to reconfigure tings around to get this behavior (totally optional tho)

Reply:
So I was thinking about how to go about that and I'm a bit uncertain. I redid my tachy method so that it doesn't need to take in either average_hr_val or tachy_limit, would that make it easier to set something like this up?